src/openapi/streaming/streaming.ts

```diff
@@ -744,11 +765,86 @@ class Streaming extends MicroEmitter<EmittedEvents> {
      * @param subscription - subscription
      */
     private onOrphanFound(subscription: Subscription) {
-        log.info(LOG_AREA, 'Subscription has become orphaned - resetting', {
-            referenceId: subscription.referenceId,
-            streamingContextId: subscription.streamingContextId,
-            servicePath: subscription.servicePath,
-        });
+        try {
+            log.info(LOG_AREA, 'Subscription has become orphaned - resetting', {
+                referenceId: subscription.referenceId,
+                streamingContextId: subscription.streamingContextId,
+                servicePath: subscription.servicePath,
+            });
+
+            const now = Date.now();
+            const lastMinute = now - 60 * 1000;
+            this.orphanEvents = this.orphanEvents.filter(
+                (orphanEvent) => orphanEvent.datetime < lastMinute,

```

Comment:
shouldn't this be `datetime > lastMinute` to get only events fired withing last minute? 🤔